clients/spark-extensions-base/src/main/scala/org/apache/spark/sql/execution/datasources/v2/BaseUseReferenceExec.scala

```diff
@@ -34,7 +34,12 @@ abstract class BaseUseReferenceExec(
   ): Seq[InternalRow] = {
 
     val ref = NessieUtils.calculateRef(branch, timestampOrHash, api)
-    NessieUtils.setCurrentRefForSpark(currentCatalog, catalog, ref)
+    NessieUtils.setCurrentRefForSpark(
+      currentCatalog,
+      catalog,
+      ref,
+      null != NessieUtils.hashFrom(timestampOrHash)
```

Comment:
`NessieUtils.hashFrom` is called two times for `timestampOrHash`

one is inside `calculateRef` and one is outside.

, call we compute it once and reuse for both ?

Reply:
I actually simplified it to just checking `timestampOrHash.isDefined`